app/Console/Commands/Schedule/CouponCommand.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Pterodactyl\Console\Commands\Schedule;
+
+use Carbon\Carbon;
+use Pterodactyl\Models\Coupon;
+use Illuminate\Console\Command;
+
+class CouponCommand extends Command
+{
+    protected $signature = 'p:schedule:coupon';
+    protected $description = 'Process coupon expirations.';
+
+    public function handle(): void
+    {
+        $this->line('Beginning check for expired coupons.');
+        $time = Carbon::now()->timestamp;
+        $coupons = Coupon::query()->get();
+        foreach ($coupons as $coupon) {
+            $carbon = new Carbon($coupon->expires);
+            $expires = $carbon->timestamp;
+            if ($time >= $expires) {
+                $coupon->update(['expired' => true]);
+                $this->line('Coupon #'.$coupon->id.' has been set as expired.');
+            }
+        }
+        $this->line('Completed check for expired coupons.');
```

Comment:
PHP-CS-Fixer will not like this spacing - also, there is no use for all of the local vars

Reply:
if ($time >= new Carbon($coupon->expires)->timestamp) { ... }

Reply:
> 

![image](https://user-images.githubusercontent.com/62303212/211694261-e717eea6-9a26-4b8f-803d-e8843800dfa4.png)
Attempted this and it failed.